cleverhans/dataset.py

```diff
@@ -0,0 +1,144 @@
+"""Dataset class for CleverHans
+
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from cleverhans.utils_mnist import data_mnist
+
+keras = None  # Only load keras if user tries to use a dataset that requires it
+
+
+class Dataset(object):
+    """Abstract base class representing a dataset.
+    """
+
+    # The number of classes in the dataset. Should be specified by subclasses.
+    NB_CLASSES = None
+
+    def __init__(self, **kwargs):
+        if "self" in kwargs:
+            del kwargs[self]
+        self.kwargs = kwargs
+
+    def get_factory(self):
+        """Returns a picklable callable that recreates the dataset.
+        """
+
+        if hasattr(self, 'kwargs'):
```

Comment:
Isn't this statement always true? If it was false it would be because the children would not have invoked the parent constructor which would be unexpected behavior. You could do `assert(hasattr(self, 'kwargs'))` to ensure the constructor was properly called.

Reply:
Also following up on your comment if the child class has no parameters, locals() will only have self and everything will work as expected.

Reply:
ok, I took out the check